numba_cuda/numba/cuda/cudadrv/nvrtc.py

```diff
@@ -251,6 +253,40 @@ def compile(src, name, cc, ltoir=False):
     nvrtc = NVRTC()
     program = nvrtc.create_program(src, name)
 
+    version = nvrtc.get_version()
+    ver_str = lambda ver : f"{ver[0]}.{ver[1]}"
+    cc_str = lambda cc: f"{cc[0]}{cc[1]}"
+    if version < (11, 0):
+        raise RuntimeError(
+            "Unsupported CUDA version. CUDA 11.0 or higher is required."
+        )
+    else:
+        try:
+            min_cc, max_cc = _CUDA_CC_MIN_MAX_SUPPORT[version]
+        except KeyError:
+            min_cc, max_cc = _CUDA_CC_MIN_MAX_SUPPORT.values()[-1]
+            max_rtc_ver = _CUDA_CC_MIN_MAX_SUPPORT.keys()[-1]
```

Comment:
Don't assume how this is sorted.
```suggestion
            max_rtc_ver, (min_cc, max_cc) = sorted(
                _CUDA_CC_MIN_MAX_SUPPORT.items(), reverse=True
            )[0]
```